Neos.Flow/Classes/Mvc/Routing/AttributeRoutesProvider.php

```diff
@@ -0,0 +1,138 @@
+{
+    /**
+     * @param array<string> $classNames
+     */
+    public function __construct(
+        public readonly ReflectionService $reflectionService,
+        public readonly ObjectManagerInterface $objectManager,
+        public readonly array $classNames,
+    ) {
+    }
+
+    public function getRoutes(): Routes
+    {
+        $routes = [];
+        $annotatedClasses = $this->reflectionService->getClassesContainingMethodsAnnotatedWith(Flow\Route::class);
+
+        foreach ($annotatedClasses as $className) {
+            $includeClassName = false;
+            foreach ($this->classNames as $classNamePattern) {
+                if (fnmatch($classNamePattern, $className, FNM_NOESCAPE)) {
+                    $includeClassName = true;
+                    break;
+                }
+            }
+            if (!$includeClassName) {
+                continue;
+            }
+            $controllerObjectName = $this->objectManager->getCaseSensitiveObjectName($className);
+            $controllerPackageKey = $this->objectManager->getPackageKeyByObjectName($controllerObjectName);
+            $controllerPackageNamespace = str_replace('.', '\\', $controllerPackageKey);
+            if (!str_ends_with($className, 'Controller')) {
+                throw new InvalidControllerException('Only for controller classes');
+            }
+
+            $localClassName = substr($className, strlen($controllerPackageNamespace) + 1);
+
+            if (str_starts_with($localClassName, 'Controller\\')) {
+                $controllerName = substr($localClassName, 11);
+                $subPackage = null;
+            } elseif (str_contains($localClassName, '\\Controller\\')) {
+                list($subPackage, $controllerName) = explode('\\Controller\\', $localClassName);
+            } else {
+                throw new InvalidControllerException('Unknown controller pattern');
+            }
+
+            $annotatedMethods = $this->reflectionService->getMethodsAnnotatedWith($className, Flow\Route::class);
+            foreach ($annotatedMethods as $methodName) {
+                if (!str_ends_with($methodName, 'Action')) {
+                    throw new InvalidActionNameException('Only for action methods');
+                }
+                $annotations = $this->reflectionService->getMethodAnnotations($className, $methodName, Flow\Route::class);
+                foreach ($annotations as $annotation) {
+                    if ($annotation instanceof Flow\Route) {
+                        $controller = substr($controllerName, 0, -10);
+                        $action = substr($methodName, 0, -6);
+                        $defaults = array_filter(
+                            $annotation->defaults ?? [],
+                            fn($key) => !in_array($key, ['@package', '@subpackage', '@controller', '@action']),
+                            ARRAY_FILTER_USE_KEY
+                        );
```

Comment:
I would suggest to detect those directly in the `Route` attribute implementation and throw.
This will simplify this code and notify users about the mis-usage

Reply:
Yes even better than what i suggested with https://github.com/neos/flow-development-collection/pull/3325#discussion_r1535954865

Reply:
Implement that